Source/Events/EventIdentifier.cs

```diff
@@ -0,0 +1,118 @@
+    /// </summary>
+    /// <remarks>
+    /// The identifier is constructed from the <see cref="MicroserviceId"/> and <see cref="TenantId"/> that committed the event, and the <see cref="EventLogSequenceNumber"/> it got in the event log.
+    /// </remarks>
+    public class EventIdentifier : ConceptAs<string>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventIdentifier"/> class.
+        /// </summary>
+        /// <param name="microserviceId">The <see cref="MicroserviceId"/> that committed the event.</param>
+        /// <param name="tenantId">The <see cref="TenantId"/> that committed the event.</param>
+        /// <param name="sequenceNumber">The <see cref="EventLogSequenceNumber"/> of the event that was committed.</param>
+        public EventIdentifier(MicroserviceId microserviceId, TenantId tenantId, EventLogSequenceNumber sequenceNumber)
+            => Value = Compose(microserviceId, tenantId, sequenceNumber);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventIdentifier"/> class.
+        /// </summary>
+        /// <param name="eventIdentifier">The event identifier <see cref="string"/>.</param>
+        public EventIdentifier(string eventIdentifier)
+        {
+            ThrowIfEventIdentifierIsInvalid(eventIdentifier);
+            Value = eventIdentifier;
+        }
+
+        /// <summary>
+        /// Composes an event identifier <see cref="string"/> from its components.
+        /// </summary>
+        /// <param name="microservice">The <see cref="MicroserviceId"/> that committed the event.</param>
+        /// <param name="tenant">The <see cref="TenantId"/> that committed the event.</param>
+        /// <param name="sequenceNumber">The <see cref="EventLogSequenceNumber"/> of the event that was committed.</param>
+        /// <returns>An event identifier <see cref="string"/>.</returns>
+        /// <remarks>
+        /// The event identifier string is the Base64 encoding of the concatenated bytes of the microservice ++ tenant ++ event log sequence number (40 bytes total).
+        /// When converting the values to bytes, they are in the order that they are transmitted over gRPC as defined by the contracts - i.e. little endian.
+        /// </remarks>
+        public static string Compose(MicroserviceId microservice, TenantId tenant, EventLogSequenceNumber sequenceNumber)
+        {
+            var microserviceBytes = microservice.Value.ToByteArray();
+            var tenantBytes = tenant.Value.ToByteArray();
+            var sequenceNumberBytes = BitConverter.GetBytes(sequenceNumber.Value);
+
+            if (!BitConverter.IsLittleEndian)
+                Array.Reverse(sequenceNumberBytes);
+
+            using var stream = new MemoryStream(40);
+
+            stream.Write(microserviceBytes);
+            stream.Write(tenantBytes);
+            stream.Write(sequenceNumberBytes);
+
+            return Convert.ToBase64String(stream.ToArray());
```

Comment:
Are we sure that we don't accidentally mess with the casing of this string and break the base64 encoding? Asking just to be sure.

Reply:
This is just copy pasta code, @jakhog was the author. I don't even know if we need this anymore?